notes/zrq/20220725-04-gaiadmpsetup.txt

```diff
@@ -0,0 +1,239 @@
+    >     Branch '20220725-zrq-include-neighbours' set up to track remote branch '20220725-zrq-include-neighbours' from 'origin'.
+    >   Installing collected packages: gaiadmpsetup
+    >     Found existing installation: gaiadmpsetup 0.1.1
+    >       Uninstalling gaiadmpsetup-0.1.1:
+    >         Successfully uninstalled gaiadmpsetup-0.1.1
+    >     Running setup.py install for gaiadmpsetup: started
+    >       Running setup.py install for gaiadmpsetup: finished with status 'done'
+    >   Successfully installed gaiadmpsetup-0.1.1
+    >   
+    >   Node [worker03]
+    >   Mon Jul 25 14:56:59 UTC 2022
+    >   iris-gaia-red-20220725-worker03
+    >   WARNING: Running pip install with root privileges is generally not a good idea. Try `pip install --user` instead.
+    >   Collecting git+https://github.com/Zarquan/gaiadmpsetup@20220725-zrq-include-neighbours
+    >     Cloning https://github.com/Zarquan/gaiadmpsetup (to revision 20220725-zrq-include-neighbours) to /tmp/pip-req-build-_0ti4gtx
+    >     Running command git clone -q https://github.com/Zarquan/gaiadmpsetup /tmp/pip-req-build-_0ti4gtx
+    >     Running command git checkout -b 20220725-zrq-include-neighbours --track origin/20220725-zrq-include-neighbours
+    >     Switched to a new branch '20220725-zrq-include-neighbours'
+    >     Branch '20220725-zrq-include-neighbours' set up to track remote branch '20220725-zrq-include-neighbours' from 'origin'.
+    >   Installing collected packages: gaiadmpsetup
+    >     Found existing installation: gaiadmpsetup 0.1.1
+    >       Uninstalling gaiadmpsetup-0.1.1:
+    >         Successfully uninstalled gaiadmpsetup-0.1.1
+    >     Running setup.py install for gaiadmpsetup: started
+    >       Running setup.py install for gaiadmpsetup: finished with status 'done'
+    >   Successfully installed gaiadmpsetup-0.1.1
+
+
+# -----------------------------------------------------
+# Restart Zeppelin
+#[root@ansibler]
+
+    ssh zeppelin \
+        '
+        zeppelin-daemon.sh restart
+        '
+
+    >   ....
+    >   Fail to execute line 3: import gaiadmpsetup
+    >   Traceback (most recent call last):
+    >     File "/tmp/1658763842028-0/zeppelin_python.py", line 158, in <module>
+    >       exec(code, _zcUserQueryNameSpace)
+    >     File "<stdin>", line 3, in <module>
+    >     File "/usr/local/lib/python3.7/site-packages/gaiadmpsetup/__init__.py", line 1, in <module>
+    >       from . import gaiadmpsetup
+    >     File "/usr/local/lib/python3.7/site-packages/gaiadmpsetup/gaiadmpsetup.py", line 6, in <module>
+    >       from . import gaiadr3_pyspark_schema_structures as dr3
+    >     File "/usr/local/lib/python3.7/site-packages/gaiadmpsetup/gaiadr3_pyspark_schema_structures.py", line 1671, in <module>
+    >       ([tmasspscxsc_best_neighbour_schema, twomass_psc_schema], release_folder + '/GDR3_2MASSPSC_BEST_NEIGHBOURS'),
+    >   NameError: name 'twomass_psc_schema' is not defined
+    >   ....
```

Comment:
Does this error mean we need to run some more tests before merging into master?

Reply:
> Does this error mean we need to run some more tests before merging into master?

These notes were looking at solving issue #909. Taking a clone of gaiadmpsetup and trying to update it to include the extra tables. Turns out way more complicated that expected, partly because I didn't have a good handle on how the gaiadmpsetup code works.

Issue #909 is optional.

The current schema works for all the existing examples. Users can use the neighbour tables from eDR3 by using the catalog prefix (which is what Nigel's howto recommends).

IF it was easy to fix, this would be a nice thing to make it easier to use. It turned out to be not that simple so it becomes a separate epic and we solve it later.

Reply:
> Had a go at adding the extra tables in a clone gaiadmpsetup, became more complicated than can be done in a couple hours. Leave this for another day.
https://github.com/wfau/aglais/issues/909#issuecomment-1194262288

Reply:
We don't even know if Nigel wants the external catalogs included in DR3. So this should probably wait until he gets bacl.

Reply:
> As far as I know, we don't have any notebooks that use the external tables. Which probably means that they have never been tested.
issue #910 

I don't think anyone has ever used the external tables. So no science case for including them.